version: '3'

volumes:
  gpapi-python-site-packages: {}
  gpapi-python-dist-packages: {}

services:
  django:
    container_name: gpapi-django
    build:
      context: .
      dockerfile: Dockerfile-django
    depends_on:
      - postgresql
    links:
      - postgresql
    command: /gpapi-django-start.sh
    ports:
      - '8000:8000'
    volumes:
      - ./:/app
      - gpapi-python-site-packages:/usr/local/lib/python3.6/site-packages
      - gpapi-python-dist-packages:/usr/local/lib/python3.6/dist-packages
    environment:
      - SECRET_KEY=a6lbgko!k@_5jv%@tqkl0ph#0dc)1zyu!41b0ehtyma93obes8
      - DEBUG=true
      - POSTGRES_HOST=postgresql
      - POSTGRES_NAME=grassparty
      - POSTGRES_USER=grassparty
      - POSTGRES_PASSWORD=grasspartypassword

  postgresql:
    container_name: gpapi-postgresql
    image: postgres:9.6
    volumes:
      - ./postgresql/data:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=grassparty
      - POSTGRES_USER=grassparty
      - POSTGRES_PASSWORD=grasspartypassword
      - POSTGRES_INITDB_ARGS=--encoding=UTF-8
    healthcheck:
      test: "pg_isready -h localhost -p 5432 -q -U postgres"
      interval: 3s
      timeout: 1s
      retries: 10
